core/src/main/java/org/apache/accumulo/core/data/Mutation.java

```diff
@@ -772,6 +772,236 @@ public void putDelete(byte[] columnFamily, byte[] columnQualifier, ColumnVisibil
     put(columnFamily, columnQualifier, columnVisibility.getExpression(), true, timestamp, true, EMPTY_BYTES);
   }
 
+  /**
+   * Fluent API for Mutation
+   *
+   * <p>
+   * Each step of the Mutation is represented by an interface. Inheritance order ensures that chained methods follow prescribed order of:
+   *
+   * <p>
+   * row, family, qualifier, visibility, timestamp
+   *
+   * <p>
+   * set and delete methods end the chain and finalize the mutation
+   */
+  public interface RowStep extends FamilyStep {
+    FamilyStep row(byte[] r);
+
+    FamilyStep row(CharSequence r);
+
+    FamilyStep row(ByteBuffer r);
+
+    FamilyStep row(Text t);
+  }
+
+  public interface FamilyStep extends QualifierStep {
```

Comment:
All of these interfaces will need a javadoc `@since 2.0` tag.  Do not need it on the method as they will inherit it from the class level javadoc.

Reply:
Updated the interface classes with the `@since 2.0` tag, thanks!